include/irr/asset/filters/dithering/IDither.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2020 - AnastaZIuk
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_I_DITHER_H_INCLUDED__
+#define __IRR_I_DITHER_H_INCLUDED__
+
+#include "irr/core/core.h"
+#include "irr/asset/ICPUImage.h"
+#include "irr/asset/ICPUImageView.h"
+
+namespace irr
+{
+    namespace asset
+    {
+        class IDither
+        {
+            public:
+                virtual ~IDither() {}
+                virtual float pGet(const core::vectorSIMDu32& pixelCoord) = 0;
+
+                class IState
+                {
+                    virtual ~IState() {}
+
+                    struct ImageAsset
+                    {
+                        core::smart_refctd_ptr<asset::IAsset> asset;
```

Comment:
it relates to the above

Reply:
not sure if we use dds dithering images or whatever in future

Reply:
dont worry about this in the base class, only provide maximum extent that the dither must support.

Reply:
okay, IState will only hold `extent`
`imageData` will be moved to precomputed dither class and will contain only strides and the buffer

Reply:
why should it have a member with texel range?

Reply:
because we wanted to have max extent the dither support, so likely texel range I think, I can limit it to only extent